crates/data_components/src/http/provider.rs

```diff
@@ -0,0 +1,1323 @@
+            ))),
+            // All other errors are internal/external errors
+            Error::HttpRequest { source } => DataFusionError::External(Box::new(source)),
+            Error::InvalidUrl { source } => DataFusionError::External(Box::new(source)),
+            Error::Arrow { source } => DataFusionError::ArrowError(Box::new(source), None),
+            Error::DataFusion { source } => source,
+        }
+    }
+}
+
+#[derive(Clone)]
+struct CachedResponse {
+    content: Arc<String>,
+    cached_at: SystemTime,
+    max_age: Duration,
+    detected_format: Option<String>,
+}
+
+impl CachedResponse {
+    fn is_fresh(&self) -> bool {
+        self.cached_at
+            .elapsed()
+            .ok()
+            .is_some_and(|elapsed| elapsed < self.max_age)
+    }
+}
+
+/// A table provider that fetches data from HTTP endpoints based on path and query filters
+#[derive(Clone)]
+pub struct HttpTableProvider {
+    base_url: Url,
+    client: Client,
+    file_format: String,
+    schema: SchemaRef,
+    constraints: Constraints,
+    cache: Arc<RwLock<HashMap<String, CachedResponse>>>,
+    acceleration_enabled: bool,
+    retry_strategy: util::fibonacci_backoff::FibonacciBackoff,
+    content_type: Option<String>,
+}
+
+impl std::fmt::Debug for HttpTableProvider {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("HttpTableProvider")
+            .field("base_url", &self.base_url)
+            .field("file_format", &self.file_format)
+            .field("acceleration_enabled", &self.acceleration_enabled)
+            .finish_non_exhaustive()
+    }
+}
+
+impl HttpTableProvider {
+    #[must_use]
+    pub fn new(
+        base_url: Url,
+        client: Client,
+        file_format: String,
+        acceleration_enabled: bool,
+    ) -> Self {
+        Self {
+            base_url,
+            client,
+            file_format,
+            schema: Arc::new(Self::base_table_schema()),
+            // Mark `_path`, `_query`, and `_body` as primary key components
+            constraints: Constraints::new_unverified(vec![Constraint::PrimaryKey(vec![0, 1, 2])]),
+            cache: Arc::new(RwLock::new(HashMap::new())),
+            acceleration_enabled,
+            retry_strategy: FibonacciBackoffBuilder::new().max_retries(Some(3)).build(),
+            content_type: None,
+        }
+    }
+
+    #[must_use]
+    pub fn with_max_retries(mut self, max_retries: u32) -> Self {
+        self.retry_strategy = FibonacciBackoffBuilder::new()
+            .max_retries(Some(max_retries as usize))
+            .build();
+        self
+    }
+
+    #[must_use]
+    pub fn with_content_type(mut self, content_type: Option<String>) -> Self {
+        self.content_type = content_type;
+        self
+    }
+
+    #[must_use]
+    pub fn base_table_schema() -> Schema {
+        Schema::new(vec![
+            Field::new("_path", DataType::Utf8, false),
```

Comment:
Why `_` prefix? We have these metadata-like columns in other providers without it (e.g. object store metadata in document tables).